Filter by Food Type

hibatawfeeq opened the pull request:
The user story filter by food type is completed. Once you select something from the dropdown it should filter the table and everything in it to adjust the filter. The 3 files that were edited are: index.js, index.ejs and client.js.

Please feel free to check the code and let me know if I am missing anything or need to correct anything.

Thank you so much! :)

Commit:
Filter by Food Type

The user story filter by food type is completed. Once you select something from the dropdown it should filter the table and everything in it to adjust the filter.

mdschnei commented on public/client.js:
```diff
@@ -30,4 +30,9 @@ function removeProduct() {
+
+function filterbytype() {
+      var x = document.getElementById("Types").value;
+      location.replace("http://localhost:5000/?type="+x);
```
Sorry, I realized I missed something here earlier with this, the problem being that this code will always attempt to redirect to localhost:5000, even when it is being run on the live site hosted by Heroku, which we obviously don't want. To fix this for now, please add the following to make the page redirect accordingly:
```
if (window.location.hostname === "localhost") {
    window.location.replace("http://localhost:5000/?type="+x);
  } else {
    window.location.replace("https://whatscookinggoodlooking.herokuapp.com/?type="+x);
  }
```

hibatawfeeq replied:
Perfect Matthew just fixed it!

Commit:
A minor error fix

Thanks @mdschnei for the headsup!

Commit:
Minor fixes!

Nicole made some changes on the database so I had to change natural join to left join.
